azure-devops/azext_devops/dev/common/services.py

```diff
@@ -152,9 +152,12 @@ def get_token_from_az_logins(organization, pat_token_present):
     return ''
 
 
-def get_token_from_az_login(profile, user, tenant):
+def get_token_from_az_login(profile, tenant):
```

Comment:
`get_raw_token` supports **user**, **service principal**, **managed identity** accounts.

By removing `user` from `get_token_from_az_login`, `get_raw_token` will only return token for the currently selected account:

```
1. sub1, user1, from tenant1 - default
2. sub2, user2, from tenant2
```

This will cause `get_raw_token` to get a token for `user1` in `tenant2`.

Consider using `subscription` as the key for `get_raw_token`.

Reply:
Tokens are independent of subscriptions. as ADO accounts are not linked to any azure subscriptions.

Reply:
I understand, but Azure CLI uses subscription as the primary key. If you don't specify `subscription` as the key for `get_raw_token`, matched `user` and `tenant` may occur.

Reply:
The old implementation of `get_access_token_for_resource` doesn't work for service principal an managed identity, that's why it is not recommended and removed in the MSAL migration:

https://github.com/Azure/azure-cli/blob/14cc787d0f58bc649d402b486fdecc5625eee9ac/src/azure-cli-core/azure/cli/core/_profile.py#L531-L535

```py
    def get_access_token_for_resource(self, username, tenant, resource):
        tenant = tenant or 'common'
        _, access_token, _ = self._creds_cache.retrieve_token_for_user(
            username, tenant, resource)
        return access_token
```

Reply:
User/tenant isolation was proposed in https://github.com/Azure/azure-cli/issues/15005. After that, it will make sense to make `get_raw_token` support user+tenant.